migration/identity-migration/src/main/java/io/camunda/migration/identity/config/ManagementIdentityConnector.java

```diff
@@ -7,27 +7,104 @@
 import io.camunda.migration.identity.midentity.ManagementIdentityClient;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.DefaultResponseErrorHandler;
 
 @Configuration
 public class ManagementIdentityConnector {
 
-  @Bean // will be closed automatically
+  @Bean
+  public Identity identity(final IdentityMigrationProperties identityMigrationProperties) {
+    final var properties = identityMigrationProperties.getManagementIdentity();
+    final var configuration =
+        new IdentityConfiguration(
+            properties.getBaseUrl(),
+            properties.getIssuerBackendUrl(),
+            properties.getIssuerBackendUrl(),
+            properties.getClientId(),
+            properties.getClientSecret(),
+            properties.getAudience(),
+            properties.getIssuerType());
+    return new Identity(configuration);
+  }
+
+  @Bean
   public ManagementIdentityClient managementIdentityClient(
       final RestTemplateBuilder builder,
-      final IdentityMigrationProperties identityMigrationProperties) {
+      final IdentityMigrationProperties identityMigrationProperties,
+      final Identity identity) {
     final var properties = identityMigrationProperties.getManagementIdentity();
     final var restTemplate =
         builder
             .rootUri(properties.getBaseUrl())
-            .defaultHeader("Authorization", String.format("Bearer %s", properties.getM2mToken()))
+            .interceptors(new M2MTokenInterceptor(identity, properties.getAudience()))
+            .errorHandler(new CustomErrorHandler())
             .defaultHeader("Content-Type", "application/json")
             .defaultHeader("Accept", "application/json")
             .build();
     return new ManagementIdentityClient(
         restTemplate, identityMigrationProperties.getOrganizationId());
   }
+
+  public static final class M2MTokenInterceptor implements ClientHttpRequestInterceptor {
+
+    final Identity identity;
+    final String audience;
+
+    public M2MTokenInterceptor(final Identity identity, final String audience) {
+      this.identity = identity;
+      this.audience = audience;
+    }
+
+    @Override
+    public ClientHttpResponse intercept(
+        final HttpRequest request, final byte[] body, final ClientHttpRequestExecution execution)
+        throws IOException {
+      final String token = identity.authentication().requestToken(audience).getAccessToken();
```

Comment:
:question: Shouldn't we cache this somehow? When a couple of SaaS clusters are updating at the same time, we don't want to spam the management identity with token requests for every single request we make, right?

Reply:
identity sdk is managing the cache itself.